cmake/autopep8.cmake

```diff
@@ -0,0 +1,100 @@
+#
+# CMake module to C++ static analysis against
+# Google C++ Style Guide (https://google.github.io/styleguide/cppguide.html)
+#
+# For more detials please follow links:
+#
+# - https://github.com/google/styleguide
+# - https://pypi.python.org/pypi/cpplint
+# - https://github.com/theandrewdavis/cpplint
+#
+# Copyright (c) 2016 Piotr L. Figlarek
+#
+# Usage ----- Include this module via CMake include(...) command and then add
+# each source directory via introduced by this module
+# cpplint_add_subdirectory(...) function. Added directory will be recursivelly
+# scanned and all available files will be checked.
+#
+# Example
+# -------
+# # include CMake module
+# include(cmake/cpplint.cmake)
+#
+# # add all source code directories
+# cpplint_add_subdirectory(core)
+# cpplint_add_subdirectory(modules/c-bind)
+#
+# License (MIT)
+# -------------
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# target to run autopep8.py for all configured sources
+set(AUTOPEP8_TARGET prettypython CACHE STRING "Name of python autoformatter")
```

Comment:
why do you cache this variable ? same for the other AUTOPEP8 variable and CLANG_FORMAT.
Also caching them tend flood the ccmake list of option so I would either not cache them or rename them so that more important variables appear first. What do you think ?

Reply:
No specific reason. I basically adapted what Till had adapted for the linter. It was already in there, so I kept it to be compliant with existing stuff.

As for the flooding: I agree, more important stuff should be further up. But  I think it should be there. I know you are not (or at least were not) using the ccmake to configure. But I like the interface to get a choice of options. I do not want to dig very deep in makefiles for a quick overview.

What do you think?

Reply:
So how about a prefix in front of such cached variables ? Like Z_ so that it appears at the bottom.

Reply:
Sounds like a good alternative. Did not think of it.
I will do that.

Reply:
Ok, so I cleaned it up. I also marked the linter as advanced. It is optional and has nothing to do with the compilation in the first place

What do you think about the not-advanced-appearance?